lib/api/api.js

```diff
@@ -184,14 +186,107 @@ const api = {
             }
             return { returnTagCount, isImplicitDeny };
         }
+        let formDataParser;
+        let fileEventData = null;
+
+        if (apiMethod === 'objectPost') {
+            formDataParser = busboy({ headers: request.headers });
+        }
 
         return async.waterfall([
+            next => {
+                if (apiMethod === 'objectPut' || apiMethod === 'objectPutPart') {
+                    return next(null);
+                }
+                if (apiMethod === 'objectPost' && request.headers['content-type'].includes('multipart/form-data')) {
+                    writeContinue(request, response);
+
+                    request.formData = {};
+                    let totalFieldSize = 0;
+                    const MAX_FIELD_SIZE = 20 * 1024; // 20KB
+                    formDataParser.on('field', (fieldname, val) => {
+                        totalFieldSize += Buffer.byteLength(val, 'utf8');
+                        if (totalFieldSize > MAX_FIELD_SIZE) {
+                            const err = errors.MaxPostPreDataLengthExceeded;
+                            formDataParser.emit('error', err);
+                        }
```

Comment:
AWS ignore fields below the file. If I add big fields after file it should not fail, but here it fails.

Also there is no need to save those fields after file, they can be discarded.
Maybe the `file` event could remove the `field` listener if we don't need to read them and to be sure to not emit an error

Reply:
Also why have a variable `const err` here, we can directly emit the error, or don't use the busboy writable error event but handle the request error immediately

Reply:
As I mentioned in private conversation, it seems to me that fields after file are evaluated to some extent:
I was able to test a large file which required streaming against AWS followed by another file.
The request was rejected after a certain time, stating there were too many files, and the large file was not stored.
I have the feeling the parsing will need to be reworked where:
- the first fields are evaluated for eventual authorization; I can confirm from a hand test that AWS depends on the file being the last in order, as you say ignoring the following headers
- the file is written to a temp file so as to reach the end of the stream
- any further stream data passes some checks (eg: are there 2 files?) before commiting to writing the object to storage. The extent of this post-first-file field checks needs to be examined: so far, what I'm seeing is anyfield that is not another file is ignored entirely.

Reply:
Reworked as noted above